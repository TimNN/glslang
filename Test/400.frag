#version 330 core

in vec2 c2D;
flat in int i;
out vec4 outp;
uniform sampler2D arrayedSampler[5];

void main()
{
    vec4 v;
    v = texture(arrayedSampler[i], c2D);
    outp.x = gl_ClipDistance[1];
}
