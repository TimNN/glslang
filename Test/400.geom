#version 400 core

void main()
{
    EmitStreamVertex(1);
    EndStreamPrimitive(0);
    EmitVertex();
    EndPrimitive();

    int id = gl_InvocationID;
}
