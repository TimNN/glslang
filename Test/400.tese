#version 400 core

layout(vertices = 4) out; // ERROR
layout(quads, cw) in;
layout(triangles) in;     // ERROR
layout(isolines) in;      // ERROR

layout(cw) in;
layout(ccw) in;           // ERROR

layout(fractional_odd_spacing) in;    
layout(equal_spacing) in;              // ERROR
layout(fractional_even_spacing) in;    // ERROR

layout(point_mode) in;

void main()
{
    barrier(); // ERROR

    int a = gl_MaxTessEvaluationInputComponents +
            gl_MaxTessEvaluationOutputComponents +
            gl_MaxTessEvaluationTextureImageUnits +
            gl_MaxTessEvaluationUniformComponents +
            gl_MaxTessPatchComponents +
            gl_MaxPatchVertices +
            gl_MaxTessGenLevel;

    vec4 p = gl_in[1].gl_Position;
    float ps = gl_in[1].gl_PointSize;
    float cd = gl_in[1].gl_ClipDistance[2];

    int pvi = gl_PatchVerticesIn;
    int pid = gl_PrimitiveID;
    vec3 tc = gl_TessCoord;
    float tlo = gl_TessLevelOuter[3];
    float tli = gl_TessLevelInner[1];

    gl_Position = p;
    gl_PointSize = ps;
    gl_ClipDistance[2] = cd;
}
