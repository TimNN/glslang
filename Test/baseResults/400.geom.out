400.geom
Warning, version 400 is not yet complete; some version-specific features are present, but many are missing.
ERROR: 0:13: 'invocations' : can only apply to a standalone qualifier 
ERROR: 0:20: 'patch' : not supported in this stage: geometry
ERROR: 0:20: 'gl_PointSize' : cannot add layout to redeclared block member 
ERROR: 0:20: 'gl_PointSize' : cannot add patch to redeclared block member 
ERROR: 0:25: 'length' :  array must first be sized by a redeclaration or layout qualifier
ERROR: 0:36: 'length' :  array must first be sized by a redeclaration or layout qualifier
ERROR: 0:40: 'triangles' : inconsistent input primitive for array size colorBad
ERROR: 0:44: 'triangles' : inconsistent input primitive for array size colorbad2
ERROR: 0:56: 'location' : repeated use of location 4
ERROR: 0:58: 'patch' : not supported in this stage: geometry
ERROR: 0:59: 'patch' : not supported in this stage: geometry
ERROR: 11 compilation errors.  No code generated.


invocations = 4
max_vertices = 127
input primitive = triangles
output primitive = none
ERROR: node is still EOpNull!
0:3  Function Definition: main( (void)
0:3    Function Parameters: 
0:5    Sequence
0:5      EmitStreamVertex (void)
0:5        Constant:
0:5          1 (const int)
0:6      EndStreamPrimitive (void)
0:6        Constant:
0:6          0 (const int)
0:7      EmitVertex (void)
0:8      EndPrimitive (void)
0:10      Sequence
0:10        move second child to first child (int)
0:10          'id' (int)
0:10          'gl_InvocationID' (in int)
0:23  Function Definition: foo( (void)
0:23    Function Parameters: 
0:25    Sequence
0:25      Constant:
0:25        1 (const int)
0:26      gl_Position: direct index for structure (4-component vector of float)
0:26        direct index (block{gl_Position,gl_PointSize})
0:26          'gl_in' (in 3-element array of block{gl_Position,gl_PointSize})
0:26          Constant:
0:26            1 (const int)
0:26        Constant:
0:26          0 (const int)
0:34  Function Definition: foo2( (void)
0:34    Function Parameters: 
0:36    Sequence
0:36      Constant:
0:36        1 (const int)
0:37      Constant:
0:37        3 (const int)
0:46  Function Definition: foo3( (void)
0:46    Function Parameters: 
0:48    Sequence
0:48      Constant:
0:48        3 (const int)
0:49      Constant:
0:49        3 (const int)
0:50      Constant:
0:50        3 (const int)
0:51      Constant:
0:51        3 (const int)
0:?   Linker Objects
0:?     '__anon__0' (layout(stream=0 ) out block{a})
0:?     'gl_in' (in 3-element array of block{gl_Position,gl_PointSize})
0:?     'color' (in 3-element array of 4-component vector of float)
0:?     'color2' (in 3-element array of 4-component vector of float)
0:?     'colorS' (in 3-element array of 4-component vector of float)
0:?     'colorBad' (in 4-element array of 4-component vector of float)
0:?     'colorbad2' (in 2-element array of 4-component vector of float)
0:?     'cva' (layout(location=4 ) in 3-element array of 4-component vector of float)
0:?     'cvb' (layout(location=5 ) in 3-element array of 4-component vector of float)
0:?     'cmc' (layout(location=2 ) in 3-element array of 3X3 matrix of float)
0:?     'patchIn' (patch in 4-component vector of float)
0:?     'patchOut' (layout(stream=0 ) patch out 4-component vector of float)


Linked geometry stage:

ERROR: Linking geometry stage: At least one shader must specify an output layout primitive

invocations = 4
max_vertices = 127
input primitive = triangles
output primitive = none

